test/e2e/nodepool_set_autorepair_test.go

```diff
@@ -0,0 +1,141 @@
+					UpgradeType: hyperv1.UpgradeTypeReplace,
+					AutoRepair:  true,
+				},
+				ClusterName: guestCluster.Name,
+				Replicas:    &oneReplicas,
+				Release: hyperv1.Release{
+					Image: guestCluster.Spec.Release.Image,
+				},
+				Platform: hyperv1.NodePoolPlatform{
+					Type: guestCluster.Spec.Platform.Type,
+					AWS:  awsNPInfo,
+				},
+			},
+		}
+
+		// Create NodePool for current test
+		err = mgmtClient.Create(ctx, nodePool)
+		if err != nil {
+			if !errors.IsAlreadyExists(err) {
+				t.Fatalf("failed to create nodePool %s with Autorepair function: %v", nodePool.Name, err)
+			}
+
+			// Update NodePool
+			err = mgmtClient.Get(ctx, crclient.ObjectKeyFromObject(nodePool), nodePool)
+			g.Expect(err).NotTo(HaveOccurred(), "failed getting existant nodepool")
+			np := nodePool.DeepCopy()
+			nodePool.Spec.Replicas = &oneReplicas
+			nodePool.Spec.Management.AutoRepair = true
+			if err := mgmtClient.Patch(ctx, nodePool, crclient.MergeFrom(np)); err != nil {
+				t.Fatalf("failed to update NodePool %s with Autorepair function: %v", nodePool.Name, err)
+			}
+			g.Expect(err).NotTo(HaveOccurred(), "failed to Update existant NodePool")
+		}
+
+		numZones := int32(len(clusterOpts.AWSPlatform.Zones))
+		clusterOpts.NodePoolReplicas = 1
+		numNodes := clusterOpts.NodePoolReplicas * numZones
+
+		t.Logf("Waiting for Nodes %d\n", numNodes)
+		nodes := e2eutil.WaitForNReadyNodesByNodePool(t, ctx, guestClient, numNodes, guestCluster.Spec.Platform.Type, nodePool.Name)
+		t.Logf("Desired replicas available for nodePool: %v", nodePool.Name)
+
+		// Terminate one of the machines belonging to the cluster
+		t.Log("Terminating AWS Instance with a autorepare NodePool")
+		nodeToReplace := nodes[0].Name
+		awsSpec := nodes[0].Spec.ProviderID
+		g.Expect(len(awsSpec)).NotTo(BeZero())
+		instanceID := awsSpec[strings.LastIndex(awsSpec, "/")+1:]
+		t.Logf("Terminating AWS instance: %s", instanceID)
+		ec2client := ec2Client(clusterOpts.AWSPlatform.AWSCredentialsFile, clusterOpts.AWSPlatform.Region)
+		_, err = ec2client.TerminateInstances(&ec2.TerminateInstancesInput{
+			InstanceIds: []*string{aws.String(instanceID)},
+		})
+		g.Expect(err).NotTo(HaveOccurred(), "failed to terminate AWS instance")
+
+		// Wait for nodes to be ready again, without the node that was terminated
+		t.Logf("Waiting for %d available nodes without %s", numNodes, nodeToReplace)
+		err = wait.PollUntil(30*time.Second, func() (done bool, err error) {
+			nodes := e2eutil.WaitForNReadyNodesByNodePool(t, ctx, guestClient, numNodes, guestCluster.Spec.Platform.Type, nodePool.Name)
+			for _, node := range nodes {
+				if node.Name == nodeToReplace {
+					return false, nil
+				}
+			}
+			return true, nil
+		}, ctx.Done())
+		g.Expect(err).NotTo(HaveOccurred(), "failed to wait for new node to become available")
+
+		// Test Finished. Scalling down the NodePool to void waste resources
+		err = mgmtClient.Get(ctx, crclient.ObjectKeyFromObject(nodePool), nodePool)
```

Comment:
any reason to not just delete it?
Also this should run in a defer() func so when the test fail we delete the NodePool as well. On the other side we want to make sure we don't loose dumped info.

Reply:
I've thinking a bit on that, I'm scaling them down in order to maintain the NodePool "alive". If the test fails for some reason we can check the artifacts in Prow in a more easy way. But if even deleting them they will appear in the DebugLog or artifacts, no issues to delete it.

Reply:
I've created a generic function to downScale the nodePools meanwhile we can discuss about the best solution. If in the end we decide to go with NodePoolDeletion, I will change it :).